### DISCUSSION TITLE
Why is pasting into `.java` files so slow?

### DISCUSSION THREAD

**Discussion by guw** [redhat-developer/vscode-java#3511] [Q&A]
I noticed that pasting into `.java` files is not straightforward with JDTLS and the extension.

For example:
```
String someVar = Optional.ofNullable(something).orElse(whatever);
```

Then doubleclick `whatever` to ensure it's selected. When selected hit Cmd+V and then you will see a spinning wheel behind `whatever` but the pasted value does not get inserted. This is a very poor experience.

Note, I do not see that problem in Eclipse. Thus, I suspect something must be happening here on the VS Code side.

**Comment by snjeza** [comment]
@guw I can't reproduce the issue. Could you attach a project example reproducing he error?

**Reply by guw** [reply]
We cannot because the problem is not consistent. Sometime it's fast (instant) and sometimes it's slow. The line above was just and example. Is there a code path that is executed after paste? What are "paste" handlers and how are they executed?

**Comment by snjeza** [comment]
> What are "paste" handlers and how are they executed?

It is a workspace command.

>  Is there a code path that is executed after paste?

didChange , codeAction, codeLens, foldingRange, documentSymbol...

> We cannot because the problem is not consistent. Sometime it's fast (instant) and sometimes it's slow

I suppose it depends on the project, edited class, extensions...
What extensions are you using?

**Reply by guw** [reply]
@snjeza I think I have an observation. When the paste operation is delayed (with the spinning wheel), the JDTLS seems to be "building". That's very unfortunate.

The build seems to be not cancelled quickly enough.

**Comment by snjeza** [comment]
@guw Could you show your extensions and your settings.json?  I still can't reproduce the issue.
